cpp/turbodbc/Library/src/time_helpers.cpp

```diff
@@ -5,80 +5,131 @@
 #endif
 #include <cstring>
 #include <sql.h>
-
-#include <boost/date_time/gregorian/gregorian_types.hpp>
-#include <boost/date_time/posix_time/posix_time_types.hpp>
+#include <chrono>
 
 namespace turbodbc {
 
-namespace {
-    boost::posix_time::ptime const timestamp_epoch({1970, 1, 1}, {0, 0, 0, 0});
-    boost::gregorian::date const date_epoch(1970, 1, 1);
-}
+// Alias a time_point with microsecond precision.
+using sys_time_us = std::chrono::time_point<std::chrono::system_clock, std::chrono::microseconds>;
 
+// Define the epoch for timestamps and dates.
+constexpr sys_time_us timestamp_epoch = 
+    sys_time_us(std::chrono::time_point_cast<std::chrono::microseconds>(
+        std::chrono::sys_days{std::chrono::year{1970}/1/1}
+    ));
+constexpr std::chrono::sys_days date_epoch = std::chrono::sys_days{std::chrono::year{1970}/1/1};
 
 int64_t timestamp_to_microseconds(char const * data_pointer)
 {
-    auto & sql_ts = *reinterpret_cast<SQL_TIMESTAMP_STRUCT const *>(data_pointer);
-    intptr_t const microseconds = sql_ts.fraction / 1000;
-    boost::posix_time::ptime const ts({static_cast<unsigned short>(sql_ts.year), sql_ts.month, sql_ts.day},
-            {sql_ts.hour, sql_ts.minute, sql_ts.second, microseconds});
-    return (ts - timestamp_epoch).total_microseconds();
+    auto const & sql_ts = *reinterpret_cast<SQL_TIMESTAMP_STRUCT const *>(data_pointer);
+
+    // Build the date component using C++20 calendar types.
+    std::chrono::year_month_day ymd{
+        std::chrono::year{sql_ts.year},
+        std::chrono::month{sql_ts.month},
+        std::chrono::day{sql_ts.day}
+    };
+    std::chrono::sys_days date_part{ymd};
+
+    // Build the time-of-day portion.
+    // Note: sql_ts.fraction is assumed to be in nanoseconds.
+    auto micros_from_fraction = std::chrono::microseconds(sql_ts.fraction / 1000);
```

Comment:
[std::chrono::nanoseconds](https://en.cppreference.com/w/cpp/chrono/duration) seems like a better fit given the input type, and this is time_point_cast<> to microseconds below anyway.

Reply:
Thanks, updated!